### PR TITLE
storage: add kv metrics.

### PR DESCRIPTION
**Pull request by disksing**
PTAL @ngaut @siddontang @zhangjinpeng1987 @overvenus

### CODE REVIEW

File: src/storage/metrics.rs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        register_gauge!(
+            "tikv_scheduler_contex_total",
+            "Total number of pending commands."
+        ).unwrap();
+
+    pub static ref SCHED_WORKER_COUNTER_VEC: CounterVec =
+        register_counter_vec!(
+            "tikv_scheduler_worker_command_total",
+            "Total number of commands executed by worker pool.",
+            &["type", "read/write"]
```
Comment (overvenus) [inline src/storage/metrics.rs]: `read/write` is a invalid label name.

> Label names may contain ASCII letters, numbers, as well as underscores. They must match the regex [a-zA-Z_][a-zA-Z0-9_]*. Label names beginning with __ are reserved for internal use.

Reference:
[Metric names and labels](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels)
[best practices for naming metrics and labels](https://prometheus.io/docs/practices/naming/)

File: src/storage/mod.rs [hunk 2]
```diff
@@ -333,7 +352,9 @@ impl Storage {
             key: key,
             start_ts: start_ts,
         };
+        let tag = cmd.tag();
         try!(self.send(cmd, StorageCb::SingleValue(callback)));
+        KV_COMMAND_COUNTER_VEC.with_label_values(&[tag]).inc();
```
Comment (overvenus) [inline src/storage/mod.rs]: Why not `KV_COMMAND_COUNTER_VEC.with_label_values(&[cmd.tag()]).inc();` ?

File: src/storage/metrics.rs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        ).unwrap();
+
+    pub static ref SCHED_HISTOGRAM_VEC: HistogramVec =
+        register_histogram_vec!(
+            "tikv_scheduler_command_duration",
```
Comment (siddontang) [inline src/storage/metrics.rs]: mostly, we use `duration_seconds`.

**Comment by disksing** [comment]
PTAL @overvenus @siddontang

**Comment by overvenus** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Pull request merged**